### ISSUE TITLE
[rootless ] `kubectl run -it`:  failed to create OCI runtime console socket: stat /run/user/1000: no such file or directory

### ISSUE THREAD

**Issue by AkihiroSuda** [rancher/k3s#2421]
**Environmental Info:**
K3s Version:  v1.19.3+k3s1 (974ad30b)

Node(s) CPU architecture, OS, and Version:  Ubuntu 20.04 (kernel 5.4.0-51-generic), x86_64

Cluster Configuration: single-node rootless

**Describe the bug:**

`kubectl run -it` fails with  `failed to create OCI runtime console socket: stat /run/user/1000: no such file or directory`

**Steps To Reproduce:**

```console
$ k3s server --rootless
```

```console
$ k3s kubectl run -it --image hello-world --restart=Never hello-world
pod default/hello-world terminated (StartError)
failed to create containerd task: failed to create OCI runtime console socket: stat /run/user/1000: no such file or directory: unknown
```

**Expected behavior:**

Shouldn't cause error

**Actual behavior:**

An error is happening

**Comment by brandond** [comment]
This appears to be coming from runc by way of containerd - https://github.com/rancher/k3s/blob/master/vendor/github.com/containerd/go-runc/console.go#L54 is returning an error - it appears to be attempting to create a socket in a tempdir within XDG_RUNTIME_DIR. I was trying to figure out why this is occurring, but I can't even get that far - rootless dies with:

```
F1027 01:19:45.572665      29 server.go:261] failed to run Kubelet: failed to create kubelet: open /dev/kmsg: operation not permitted
goroutine 4313 [running]:
github.com/rancher/k3s/vendor/k8s.io/klog/v2.stacks(0xc000010001, 0xc0150c2240, 0x86, 0x212)
        /go/src/github.com/rancher/k3s/vendor/k8s.io/klog/v2/klog.go:996 +0xb9
github.com/rancher/k3s/vendor/k8s.io/klog/v2.(*loggingT).output(0x6ea03a0, 0xc000000003, 0x0, 0x0, 0xc00970e5b0, 0x6b57a48, 0x9, 0x105, 0x0)
        /go/src/github.com/rancher/k3s/vendor/k8s.io/klog/v2/klog.go:945 +0x191
github.com/rancher/k3s/vendor/k8s.io/klog/v2.(*loggingT).printDepth(0x6ea03a0, 0xc000000003, 0x0, 0x0, 0x1, 0xc006349ce0, 0x1, 0x1)
        /go/src/github.com/rancher/k3s/vendor/k8s.io/klog/v2/klog.go:718 +0x165
github.com/rancher/k3s/vendor/k8s.io/klog/v2.(*loggingT).print(...)
        /go/src/github.com/rancher/k3s/vendor/k8s.io/klog/v2/klog.go:703
github.com/rancher/k3s/vendor/k8s.io/klog/v2.Fatal(...)
        /go/src/github.com/rancher/k3s/vendor/k8s.io/klog/v2/klog.go:1436
github.com/rancher/k3s/vendor/k8s.io/kubernetes/cmd/kubelet/app.NewKubeletCommand.func1(0xc00bb7c580, 0xc0027ab400, 0x24, 0x40)
        /go/src/github.com/rancher/k3s/vendor/k8s.io/kubernetes/cmd/kubelet/app/server.go:261 +0x64d
github.com/rancher/k3s/vendor/github.com/spf13/cobra.(*Command).execute(0xc00bb7c580, 0xc0027ab400, 0x24, 0x40, 0xc00bb7c580, 0xc0027ab400)
        /go/src/github.com/rancher/k3s/vendor/github.com/spf13/cobra/command.go:846 +0x2c2
github.com/rancher/k3s/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00bb7c580, 0x0, 0x0, 0x0)
        /go/src/github.com/rancher/k3s/vendor/github.com/spf13/cobra/command.go:950 +0x375
github.com/rancher/k3s/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /go/src/github.com/rancher/k3s/vendor/github.com/spf13/cobra/command.go:887
github.com/rancher/k3s/pkg/daemons/executor.Embedded.Kubelet.func1(0xc00bb7c580)
        /go/src/github.com/rancher/k3s/pkg/daemons/executor/embed.go:31 +0x2f
created by github.com/rancher/k3s/pkg/daemons/executor.Embedded.Kubelet
        /go/src/github.com/rancher/k3s/pkg/daemons/executor/embed.go:30 +0x8d
```

There appears to be a thread about this over at https://github.com/kubernetes-sigs/kind/issues/1668#issuecomment-645878418

**Comment by AkihiroSuda** [comment]
k3s seems somehow mounting a new tmpfs on `/run` over RootlessKit's `/run`, so `/run/user` is hidden

**Comment by AkihiroSuda** [comment]
Fix: #2586
